### PR TITLE
feat: subaccount service broker

### PR DESCRIPTION
**Pull request by v0lkc**
## Purpose

* enables datasource `btp_subaccount_service_broker`
* enables datasource `btp_subaccount_service_brokers`
* implements resource `btp_subaccount_service_broker` as requested in #855 

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type

What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

* Test the code via automated test

```bash
go test ./...
```

* Additional test steps

```
...
```

## What to Check

Verify that the following are valid:

* Automated tests are executed successfully

* ...

## Checklist for reviewer

The following organizational tasks must be completed before merging this PR:

* [x] The PR is assigned to the Terraform project and a status is set (typically "in review").
* [x] The PR has the matching labels assigned to it.
* [x] The PR has a milestone assigned to it.
* [x] If the PR closes an issue, the issue is referenced.
* [x] Possible follow-up items are created and linked.

### CODE REVIEW

**Comment by v0lkc** [comment]
I couldn't make `labels` available on the `btp_subaccount_service_broker` resource because of NGPBUG-397042.

File: internal/btpcli/facade_services_broker.go [hunk 1]
```diff
@@ -47,3 +48,54 @@ func (f servicesBrokerFacade) GetByName(ctx context.Context, subaccountId string
...
+	params, err := tfutils.ToBTPCLIParamsMap(args)
+
+	if err != nil {
+		return servicemanager.ServiceBrokerResponseObject{}, CommandResponse{}, err
+	}
+
+	return doExecute[servicemanager.ServiceBrokerResponseObject](f.cliClient, ctx, NewUpdateRequest(f.getCommand(), params))
+}
+
+// TODO Update
```
Comment (lechnerc77) [inline internal/btpcli/facade_services_broker.go]: The `Update` function is available. Does this mean something different?

File: internal/provider/resource_subaccount_service_broker.go [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+
+func (rs *subaccountServiceBrokerResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan subaccountServiceBrokerResourceType
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	cliReq := btpcli.SubaccountServiceBrokerRegisterInput{
```
Comment (lechnerc77) [inline internal/provider/resource_subaccount_service_broker.go:159]: Is there no polling needed i.e., the registration is fully available after the CREATE or UPDATE command was triggered?

File: internal/provider/resource_subaccount_service_broker_test.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+						resource.TestCheckResourceAttr("btp_subaccount_service_broker.uut", "username", "platform"),
+						resource.TestCheckResourceAttr("btp_subaccount_service_broker.uut", "password", "a-secure-password"),
+						resource.TestMatchResourceAttr("btp_subaccount_service_broker.uut", "created_date", regexpValidRFC3999Format),
+						resource.TestMatchResourceAttr("btp_subaccount_service_broker.uut", "last_modified", regexpValidRFC3999Format),
+					),
+				},
+			},
+		})
+	})
+}
```
Comment (lechnerc77) [inline internal/provider/resource_subaccount_service_broker_test.go:58]: The Import of a broker should be added to the test setup

File: internal/provider/datasource_subaccount_service_broker.go
Comment (lechnerc77) [inline internal/provider/datasource_subaccount_service_broker.go:1]: Tests are missing for the service broker data sources

**Review by lechnerc77 — CHANGES_REQUESTED** [review]
Thanks for the contribution. I have some review comments that need to be addressed before we can merge it

File: internal/btpcli/facade_services_broker.go (hunk 1 above)
Comment (v0lkc) [reply internal/btpcli/facade_services_broker.go]: Good catch. I'll remove it.

File: internal/provider/resource_subaccount_service_broker.go (hunk 2 above)
Comment (v0lkc) [reply internal/provider/resource_subaccount_service_broker.go:159]: I'm not sure if it is required. But to be on the safe side we can add some logic to wait for the `ready` flag to become true.

File: internal/provider/resource_subaccount_service_broker_test.go (hunk 3 above)
Comment (v0lkc) [reply internal/provider/resource_subaccount_service_broker_test.go:58]: I'll add a test

File: internal/provider/datasource_subaccount_service_broker.go
Comment (v0lkc) [reply internal/provider/datasource_subaccount_service_broker.go:1]: Oh, i didn't notice that. I thought we'd have already some in.

Comment (lechnerc77) [reply internal/provider/datasource_subaccount_service_broker.go:1]: I was also surprised tbh

**Review by lechnerc77 — APPROVED** [review]
LGTM

**Pull request merged**